GeneralsMD/Code/GameEngine/Source/GameNetwork/LANAPIhandlers.cpp

```diff
@@ -290,7 +311,34 @@ void LANAPI::handleRequestJoin( LANMessage *msg, UnsignedInt senderIP )
 			}
 #endif
 
-			// We're the host, so see if he has a duplicate name
+			// TheSuperHackers @bugfix slurmlord 18/09/2025 need to validate the name of the connecting player before
+			// allowing them to join to prevent messing up the format of game state string. Commas, colons, semicolons etc.
+			// should not be in a player name. It should also not consist of only space characters.
+			if (canJoin)
+			{
+				const size_t nameLen = wcslen(msg->name);
+				Bool containsIllegalChars = false;
+				Bool containsNonSpaces = false;
+				for (size_t i = 0; i < nameLen; ++i)
```

Comment:
Here is more elegant way to iterate zero terminated string, without using strlen:

```cpp
static Bool PlayerNameContainsInvalidChars(const char* str)
{
  DEBUG_ASSERTCRASH(str != NULL, ("str is NULL"));
  while (*str)
  {
    if (IsInvalidCharForPlayerName(*str++))
      return true;
  }
  return false;
}
```